### PR TITLE
Add a doc page on using Address/Memory/Thread sanitizers

### PR DESCRIPTION
**Pull request by bjacob**

### CODE REVIEW

File: docs/developing_iree/sanitizers.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Using Address/Memory/Thread sanitizers
+
+## Enabling the sanitizers
+
+In the CMake build system of IREE, at least on Linux and Android, enabling these sanitizers is a simple matter of passing one of these options to the initial CMake command:
```
Comment (antiagainst) [inline docs/developing_iree/sanitizers.md:5]: Wrap these around 80 characters?

**Review by antiagainst — APPROVED** [review]

File: docs/developing_iree/sanitizers.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Using Address/Memory/Thread sanitizers
+
+## Enabling the sanitizers
```
Comment (ScottTodd) [inline docs/developing_iree/sanitizers.md:3]: Please add a bit of an introduction explaining what sanitizers are and how they are useful

you can link to these docs:
https://clang.llvm.org/docs/AddressSanitizer.html
https://clang.llvm.org/docs/MemorySanitizer.html
https://clang.llvm.org/docs/ThreadSanitizer.html

the warnings like
>Typical slowdown introduced by AddressSanitizer is 2x.

would also be good to mention (so people don't always enable the sanitizers and later complain about performance issues)

**Review by ScottTodd — APPROVED** [review]

File: docs/developing_iree/sanitizers.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# Using Address/Memory/Thread sanitizers
```
Comment (antiagainst) [inline docs/developing_iree/sanitizers.md:1]: Nit: s/sanitizers/Sanitizers/

File: docs/developing_iree/sanitizers.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+[AddressSanitizer](https://clang.llvm.org/docs/AddressSanitizer.html),
+[MemorySanitizer](https://clang.llvm.org/docs/MemorySanitizer.html) and
+[ThreadSanitizer](https://clang.llvm.org/docs/ThreadSanitizer.html) are tools
+provided by `clang` to detect certain classes of errors in C/C++ programs. They
+consist of compiler instrumentation (so your program's executable code is
+modified) and runtime libraries (so e.g. the `malloc` function may get
+replaced).
+
+They are routinely abbreviated "ASan", "MSan" and "TSan" respectively.
```
Comment (antiagainst) [inline docs/developing_iree/sanitizers.md:11]: abbreviated as?

**Review by antiagainst — APPROVED** [review]

File: docs/developing_iree/sanitizers.md (hunk 1 above)
Comment (bjacob) [reply docs/developing_iree/sanitizers.md]: done

File: docs/developing_iree/sanitizers.md (hunk 2 above)
Comment (bjacob) [reply docs/developing_iree/sanitizers.md]: done

File: docs/developing_iree/sanitizers.md (hunk 3 above)
Comment (bjacob) [reply docs/developing_iree/sanitizers.md:1]: done

File: docs/developing_iree/sanitizers.md (hunk 4 above)
Comment (bjacob) [reply docs/developing_iree/sanitizers.md:11]: done

**Review by ScottTodd — APPROVED** [review]
👍

**Pull request merged**